### PR TITLE
Added fastai, spacy, and autologging sections for #issue 3137

### PR DESCRIPTION
**Pull request by dmatrix**
Signed-off-by: Jules Damji <dmatrix@comcast.net>

## What changes are proposed in this pull request?

Add missing documentation for built-in models fastai, spacy, and autologging functionality for fastai.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -486,6 +486,28 @@ not models that implement the `scikit-learn API
 
 For more information, see :py:mod:`mlflow.lightgbm`.
 
+Spacy(``spaCy``)
```
Comment (harupy) [inline docs/source/models.rst:489]:
```suggestion
spaCy(``spaCy``)
```

File: docs/source/tracking.rst [hunk 2]
```diff
@@ -345,11 +345,29 @@ Autologging captures the following information:
...
+| fastai    | user-specified metrics | Logs optimizer data as parameters. For exmaple,  |  --           | Model checkpoints are logged to a ‘models’ directory; `MLflow Model`_ (fastai Learner model) on training end; Model summary text is logged                            |
+|           |                        | ``epochs``, ``lr``, ``opt_func``, etc            |               |                                                                                                                                                                       |
+|           |                        | Logs the parameters of the ``EarlyStopping`` and |               |                                                                                                                                                                       |
+|           |                        | ``OneCycleScheduler`` callbacks                  |               |                                                                                                                                                                       |
```
Comment (harupy) [inline docs/source/tracking.rst:366]: Can we insert links for `EarlyStoppingCallback` and `OneCycleScheduler`?

- https://docs.fast.ai/callbacks.html#EarlyStoppingCallback
- https://docs.fast.ai/callbacks.html#OneCycleScheduler

File: docs/source/models.rst (hunk 1 above)
Comment (dmatrix) [reply docs/source/models.rst:489]: @harupy We have resorted to using a capital letter in the enumerated built-in models' list, as you can see with [here](https://mlflow.org/docs/latest/models.html#built-in-model-flavors). However, the use in the description resorts to how it's referred to in `spaCy's` documentation, though. `Fastai` is also referred to `fastai` in its documentation. So if do for `spaCy`, in this enumerated list, then we ought to do it for `fastai`, both with lower case, to be consistent. 

Or we flout that be consistent as capital for the first letter in the list, for all built-in models, to be consistent with our documentation.

File: docs/source/tracking.rst (hunk 2 above)
Comment (dmatrix) [reply docs/source/tracking.rst:366]: Would it make sense to add links here or in the `[mlflow.fastai.autolog(](https://mlflow.org/docs/latest/python_api/mlflow.fastai.html#mlflow.fastai.autolog))`

Or both?

Comment (harupy) [reply docs/source/tracking.rst:366]: Screen Shot 2020-07-30 at 8 22 02

Comment (harupy) [reply docs/source/tracking.rst:366]: Yes, let's add link in docstring of `mlflow.fastai.autolog` :)

File: docs/source/models.rst (hunk 1 above)
Comment (harupy) [reply docs/source/models.rst:489]: ok let's stick to the current approach (using a capital letter in the enumerated built-in models' list)

File: docs/source/tracking.rst [hunk 3]
```diff
@@ -345,11 +345,32 @@ Autologging captures the following information:
...
+
++-----------+------------------------+----------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Framework | Metrics                | Parameters                                               | Tags          | Artifacts                                                                                                                                                             |
++-----------+------------------------+----------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| fastai    | user-specified metrics | Logs optimizer data as parameters. For exmaple,          |  --           | Model checkpoints are logged to a ‘models’ directory; `MLflow Model`_ (fastai Learner model) on training end; Model summary text is logged                            |
```
Comment (harupy) [inline docs/source/tracking.rst:366]: typo: `exmaple` -> `example`

Comment (dmatrix) [reply docs/source/tracking.rst]: Good catch! will fix it

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**